test/testdata/lsp/code_actions/move_method/class_func.rb

```diff
@@ -0,0 +1,41 @@
+# typed: strict
+# selective-apply-code-action: refactor.extract
+
+module Foo
+  extend T::Sig
+
+  sig {returns(String)}
+  def self.greeting
+         # ^^^^^^^^ apply-code-action: [A] Move method to a new module
+    'Hello'
+  end
+  
+  sig do
+    params(x: String)
+    .returns(String)
+  end
+  def name(x)
+    "#{Foo.greeting} #{x}"
+  end
+end
+
+module A
+  class B
+    extend T::Sig
+
+    sig {void}
+    def bar
+      m = Foo
+      Foo.greeting
+
+      m.greeting
+      print((Foo if true).greeting)
+    end
+
+    sig {params(m: T.class_of(Foo)).void}
+    def example3(m)
+      m.greeting
+    end
```

Comment:
Huh. I'm surprised here. How are you detecting this?

What if `m: T.class_of(FooParent)` where

```ruby
class FooParent; end
class Foo < FooParent
  def self.greeting; end
end
```

?

Reply:
>How are you detecting this?

I'm reusing the rename infra here. `addSubclassRelatedMethods` or `addDispatchRelatedMethods` add the `greeting` symbol in the rename queue, and the `m.greeting` location is recorded in the `locs()` of that symbol.

https://github.com/sorbet/sorbet/blob/f895091951e3ba59a162834299e282383acae231/main/lsp/AbstractRenamer.cc#L92-L133

Reply:
btw, `m: T.class_of(FooParent)` would have a type error, because `FooParent` doesn't have a `greeting` defined. 

```ruby
sig {params(m: T.class_of(Foo)).void}
def example3(m)
  m.greeting
    # ^ error: Method `greeting` does not exist on `T.class_of(FooParent)`
end
```

Reply:
Sorry, I think I meant to give this example:

```ruby
# typed: true
# selective-apply-code-action: refactor.extract

class Parent
  def self.only_in_parent
    # ^ apply-code-action: [A] Move method to a new module
  end

  def self.in_parent_and_child
    # ^ apply-code-action: [B] Move method to a new module
  end
end

class Child < Parent
  def self.in_parent_and_child; end
end

class Main
  extend T::Sig

  sig {params(x: T.class_of(Child)).void}
  def example(x)
    x.only_in_parent

    x.in_parent_and_child
  end
end
```

[→ View on sorbet.run](https://sorbet.run/#%23%20typed%3A%20true%0A%23%20selective-apply-code-action%3A%20refactor.extract%0A%0Aclass%20Parent%0A%20%20def%20self.only_in_parent%0A%20%20%20%20%23%20%5E%20apply-code-action%3A%20%5BA%5D%20Move%20method%20to%20a%20new%20module%0A%20%20end%0A%0A%20%20def%20self.in_parent_and_child%0A%20%20%20%20%23%20%5E%20apply-code-action%3A%20%5BB%5D%20Move%20method%20to%20a%20new%20module%0A%20%20end%0Aend%0A%0Aclass%20Child%20%3C%20Parent%0A%20%20def%20self.in_parent_and_child%3B%20end%0Aend%0A%0Aclass%20Main%0A%20%20extend%20T%3A%3ASig%0A%0A%20%20sig%20%7Bparams%28x%3A%20T.class_of%28Child%29%29.void%7D%0A%20%20def%20example%28x%29%0A%20%20%20%20x.only_in_parent%0A%0A%20%20%20%20x.in_parent_and_child%0A%20%20end%0Aend)

The one I'm particularly surprised by is the `in_parent_and_child` case. That one is weird because I would have thought that the override in `Child` would have taken precedence over the definition in `Parent`, so given that `x` has type `T.class_of(Child)`, I'm surprised that the renamer queues up that one as a location that needs to be renamed.

The other case is also worth considering, but I think that the current behavior is probably fine.